File: Websites/Cursor_AuraWear/frontend/src/api/checkout.ts
import { apiRequest, ApiError } from "./client";
import type { CartItem } from "../types/cart";

export interface CheckoutLineItem {
  product_id: number;
  name: string;
  quantity: number;
  unit_price: string;
  line_total: string;
}

export interface CheckoutResponse {
  message: string;
  order_id: number;
  invoice_number: string;
  total: string;
  new_balance: string;
  items: CheckoutLineItem[];
}

export function checkout(items: CartItem[]) {
  return apiRequest<CheckoutResponse>("/checkout", {
    method: "POST",
    body: JSON.stringify({
      items: items.map((i) => ({ product_id: i.productId, quantity: i.quantity })),
    }),
  });
}

export async function downloadInvoice(orderId: number, invoiceNumber: string) {
  const response = await fetch(`/api/orders/${orderId}/invoice`, {
    credentials: "include",
  });

  if (!response.ok) {
    throw new ApiError(
      response.status === 404 ? "Invoice not found" : "Could not download invoice",
      response.status,
    );
  }

  const blob = await response.blob();
  const url = URL.createObjectURL(blob);
  const link = document.createElement("a");
  link.href = url;
  link.download = `${invoiceNumber}.pdf`;
  document.body.appendChild(link);
  link.click();
  link.remove();
  URL.revokeObjectURL(url);
}

File: Websites/Cursor_AuraWear/backend/app/schemas/checkout.py
from decimal import Decimal

from pydantic import BaseModel, Field


class CheckoutItemRequest(BaseModel):
    product_id: int = Field(gt=0)
    quantity: int = Field(gt=0, le=99)


class CheckoutRequest(BaseModel):
    items: list[CheckoutItemRequest] = Field(min_length=1)


class CheckoutLineItem(BaseModel):
    product_id: int
    name: str
    quantity: int
    unit_price: Decimal
    line_total: Decimal


class CheckoutResponse(BaseModel):
    message: str
    order_id: int
    invoice_number: str
    total: Decimal
    new_balance: Decimal
    items: list[CheckoutLineItem]
